util/EIP-712/src/encode.rs

```diff
@@ -56,11 +56,18 @@ fn build_dependencies<'a>(message_type: &'a str, message_types: &'a MessageTypes
 			deps.insert(item);
 
 			for field in fields {
+				// check if this field is an array type
+				let field_type = if field.type_.contains('[') {
+					let index = field.type_.find('[').expect("alrady checked in if statement; qed");
```

Comment:
we could get rid of `expect` with `if let Some(index) = field.type_.find('[') {`

Reply:
ah good catch